tests/nox/test_git_nox.py

```diff
@@ -0,0 +1,72 @@
+from unittest import mock
+
+import pytest
+from git import Repo
+
+from noxfiles.git_nox import generate_tag
+
+
+class TestGitNox:
+    """
+    Tests for git nox commands and/or utilities
+    """
+
+    @pytest.fixture(scope="session")
+    def repo(self):
+        repo = Repo()
+        git_session = repo.git()
+        git_session.fetch("--force", "--tags")
+        return repo
+
+    @mock.patch("nox.Session.log")
+    @pytest.mark.parametrize(
+        "current_branch,tags,expected_tag",
+        [
+            ("main", ["2.9.0"], "2.9.1b0"),
+            ("main", ["2.9.1", "2.9.0"], "2.9.2b0"),
+            ("some-test-feature-branch", ["2.9.1"], "2.9.2a0"),
+            ("some-test-feature-branch", ["2.9.1", "2.9.0"], "2.9.2a0"),
+            (
+                "some-test-feature-branch",
+                ["2.9.3a0", "2.9.2"],
+                "2.9.3a1",
+            ),  # our repo happens to already have a 2.9.3a0 tag, let's use it for testing here
+            (
+                "some-other-feature",
+                ["2.14.1a0", "2.14.1a1", "2.14.0"],
```

Comment:
this is what you've added, right? looks good to me!

Reply:
Yep! Thanks 🙌🏽